File: DemoServer/Indexes/OrderByCompanyAndCountry.cs
using System.Linq;
using DemoServer.Entities;
using Raven.Client.Indexes;

namespace DemoServer.Indexes
{
    public class OrderByCompanyAndCountry : AbstractIndexCreationTask<Order>
    {
        public OrderByCompanyAndCountry()
        {
            Map = orders => from order in orders
                            select new
                            {
                                ShipTo_City = order.ShipTo.City,
                                ShipTo_Country = order.ShipTo.Country
                            };
        }
    }
}

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/indexes/OrderByCompanyAndCountry.java
package net.ravendb.demo.indexes;

import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;

public class OrderByCompanyAndCountry extends AbstractIndexCreationTask {

    public OrderByCompanyAndCountry() {
        map = "from order in docs.Orders " +
                "select new { " +
                "   ShipTo_City = order.ShipTo.City, " +
                "   ShipTo_Country = order.ShipTo.Country" +
                "}";
    }
}
